### PR TITLE
Rename commit command to branch and enhance

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description

Rename commit command to branch.  Enhance renamed branch command 
to display current branch if none provided.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap
3. juju branch                         <- Active branch is "master"
4. juju add-branch testme
3. juju branch                         <- Active branch is "testme"
3. juju branch master             <- Active branch set to "master"
3. check 'see also' of other branch related commands
4. juju help branch

## Documentation changes

This feature is still under development, however any docs referencing juju commit should be updated to use juju branch in the develop stream and enhancement noted.

### CODE REVIEW

**Comment by hmlanigan** [comment]
Only d467d9e needs reviewing, the other two commits are covered in PR 10345.

File: cmd/juju/model/branch.go [hunk 1]
```diff
@@ -47,15 +58,15 @@ func NewCheckoutCommand() cmd.Command {
 type checkoutCommand struct {
```
Comment (manadart) [inline cmd/juju/model/branch.go]: `branchCommand`?

File: cmd/juju/model/branch.go [hunk 2]
```diff
@@ -65,8 +76,8 @@ type CheckoutCommandAPI interface {
 // Info implements part of the cmd.Command interface.
 func (c *checkoutCommand) Info() *cmd.Info {
 	info := &cmd.Info{
-		Name:    "checkout",
-		Args:    "<branch name>",
+		Name:    "branch",
+		Args:    "[<branch name>]",
 		Purpose: checkoutSummary,
```
Comment (manadart) [inline cmd/juju/model/branch.go]: `branchSummary` and `branchDoc`?

File: cmd/juju/model/branch.go (hunk 1 above)
Comment (hmlanigan) [reply cmd/juju/model/branch.go]: yes.

File: cmd/juju/model/branch.go (hunk 2 above)
Comment (hmlanigan) [reply cmd/juju/model/branch.go]: yes

**Comment by hmlanigan** [comment]
@manadart thought I caught all the changes from checkout to branch.  Glad you found the ones I missed. Thank you.  Caught one more too making the change.

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**